src/info/channel.go.tmpl

```diff
@@ -11,21 +11,29 @@
 	{{ $channel = $args.Get 0 }}
 {{ end }}
 
-{{ $types := cslice "Text" "DM" "Voice" "Group DM" "Category" "News" "Store" }}
+{{ $types := sdict 
+	"0" "Text" "1" "DM" "2" "Voice" "3" "Group DM" "4" "Category" "5" "Announcement" 
+	"10" "Thread - Announcement" "11" "Thread - Public" "12" "Thread - Private" 
+	"13" "Stage" "14" "Directory" "15" "Forum" "16" "Media"
+}}
+{{ $type := $types.Get (json $channel.Type) }}
 {{ $nsfw := "No" }}
 {{ $parent := "*None set*" }}
 {{ if $channel.NSFW }} {{ $nsfw = "Yes" }} {{ end }}
-{{ with $channel.ParentID }} {{ $parent = printf "<#%d>" . }} {{ end }}
-{{ $createdAt := div $channel.ID 4194304 | add 1420070400000 | mult 1000000 | toDuration | (newDate 1970 1 1 0 0 0).Add }}
-{{ sendMessage nil (cembed
+{{ with $channel.ParentID }}
+	{{ $parent = printf "<#%d>" . }}
+	{{ if (getChannel .).IsForum }} {{ $type = "Thread - Forum" }} {{ end}}
+{{ end }}
```

Comment:
We already have the channel struct so let's not issue a gratuitous `getChannel` call fetching it again here. Just use a normal `if` and access `$channel.IsForum`.

Reply:
I did not think of the case where the channel given is the forum channel itself. I was thinking one of the forum thread channel would be the channel to retrieve info on.
In that case, `getChannel` is needed if the channel is a forum thread, as `.IsForum` returns false for forum threads (true for `.IsThread`); but true for the forum channel (.ParentID) itself.
I can add conditionals to determine whether the channel is a forum or forum thread as `.IsForum` returns differently for the forum channel itself and a forum thread. I think both should present the channel type as "Thread - Forum"; or maybe we separate them into "Forum" and "Forum - Thread". What do you think?

Reply:
Thanks, good catch. I read the code too hastily and misunderstood your intent. It is fine as is.